### PR TITLE
#2268 Correct a version calculation for non-master mainline branches

### LINKED ISSUE #2268: [Bug] Incorrect version calculation for non-master mainline
For a non-"master" mainline branch a version increments twice

## Expected Behavior
A version is calculated identical for any mainline branch
Test commit: https://github.com/Kirill-Maurin/GitVersion/commit/27a2220a3ad678c848582ed12c10dbdb9a58e176
1. Branch "mainline": 6.0.0
2. Branch "master": 6.0.0

## Actual Behavior
1. Branch "mainline": 7.0.0
2. Branch "master": 6.0.0

## Possible Fix
Use ```!context.CurrentBranch.IsSameBranch(mainline)``` instead hardcode ```context.CurrentBranch.FriendlyName != "master"```
https://github.com/GitTools/GitVersion/blob/ba7992f2a0c536157f1ea6c3aed19a042575c932/src/GitVersionCore/VersionCalculation/MainlineVersionCalculator.cs#L72

## Steps to Reproduce
1. Checkout test branch: https://github.com/Kirill-Maurin/GitVersion/tree/test/mainline
GitVersion: 7.0.0
2. Reset master to commit: https://github.com/Kirill-Maurin/GitVersion/commit/27a2220a3ad678c848582ed12c10dbdb9a58e176
3. Checkout master
GitVersion: 6.0.0

### PR DESCRIPTION
**Pull request by Kirill-Maurin**
Fixes #2268

### CODE REVIEW

File: src/GitVersionCore.Tests/IntegrationTests/MainlineDevelopmentMode.cs [hunk 1]
```diff
@@ -20,6 +20,24 @@ public class MainlineDevelopmentMode : TestBase
             VersioningMode = VersioningMode.Mainline
         };
 
+        [Test]
+        public void VertifyNonMasterMainlineVersionIdenticalAsMaster()
```
Comment (asbjornu) [inline src/GitVersionCore.Tests/IntegrationTests/MainlineDevelopmentMode.cs:24]:
```suggestion
        public void VerifyNonMasterMainlineVersionIdenticalAsMaster()
```

Comment (Kirill-Maurin) [reply src/GitVersionCore.Tests/IntegrationTests/MainlineDevelopmentMode.cs]: Done

**Comment by Kirill-Maurin** [comment]
> Just a simple typo. Otherwise this is good to go. :)

Done

**Comment by asbjornu** [comment]
@Kirill-Maurin, looking good! Do you want to do a rebase or should I just merge from master?

**Comment by arturcic** [comment]
@asbjornu let's merge it and we can release 5.3.3

**Comment by asbjornu** [comment]
Thank you for your contributions, @Kirill-Maurin! 🙏

**Pull request merged**